File: frontend/src/types/notification.ts
export type NotificationType = "environment" | "facial";

export type NotificationSeverity = "info" | "warning" | "critical";

export type NotificationSource =
  | "temperature"
  | "light"
  | "noise"
  | "eyesClosed"
  | "yawning";

export type AppNotification = {
  _id: string;
  userId: string;
  sessionId: string;
  type: NotificationType;
  severity: NotificationSeverity;
  source: NotificationSource;
  title: string;
  message: string;
  value: number | boolean;
  threshold: number | boolean | null;
  read: boolean;
  createdAt: string;
};

File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


NotificationType = Literal["environment", "facial"]
NotificationSeverity = Literal["info", "warning", "critical"]
NotificationSource = Literal[
    "temperature",
    "light",
    "noise",
    "eyesClosed",
    "yawning",
    "fatigue",
]


class NotificationResponse(BaseModel):
    _id: str
    userId: str
    sessionId: str
    type: NotificationType
    severity: NotificationSeverity
    source: NotificationSource
    title: str
    message: str
    value: float | bool
    threshold: float | bool | None = None
    read: bool
    createdAt: datetime